components/Activity.tsx

```diff
@@ -1,19 +1,46 @@
-import { StyleSheet, TouchableOpacity, View } from "react-native"
+import { StyleSheet, TouchableOpacity, View, Modal } from "react-native"
+import React, { useState } from "react"
 import CustomText from "@/components/ui/CustomText"
 import EditIcon from "@/components/vectors/EditIcon"
+import EditAvailability from "@/screens/EditAvailability"
 
 export default function Activity() {
+  const [isModalVisible, setIsModalVisible] = useState(false)
+  const [availabilityText, setAvailabilityText] = useState("Available")
```

Comment:
since the activity is being edited here, I think this state can be set as such
```suggestion
  const [activity, setActivity] = useState("Available")
```